### PR TITLE
feature: add TSA support when verifying authorities

### PR DESCRIPTION
**Pull request by hectorj2f**
Signed-off-by: Hector Fernandez <hector@chainguard.dev>

#### Summary

We are adding support to verify images using timestamp authorities. To test it, we use scaffolding and a working branch to install cosign v2.0.0-rc so we can use TSA verifications (https://github.com/sigstore/cosign-installer/pull/105).

#### Release Note

Add support to verify images using timeStamp authorities.

### CODE REVIEW

File: test/testdata/trustroot/e2e/with-tsa.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+---
+apiVersion: policy.sigstore.dev/v1alpha1
+kind: TrustRoot
+metadata:
+  name: my-sigstore-keys
+spec:
+  sigstoreKeys:
+    timestampAuthorities:
```
Comment (vaikas) [inline test/testdata/trustroot/e2e/with-tsa.yaml:21]:
```
  The TrustRoot "my-sigstore-keys" is invalid: spec.sigstoreKeys.certificateAuthorities: Invalid value: "null": spec.sigstoreKeys.certificateAuthorities in body must be of type array: "null"
  Error: Process completed with exit code 1.
```

Don't we still need the signing CA added? It's currently listed as 'required', so if this is a bad assumption we need to relax it in the trustroot API.

Comment (hectorj2f) [reply test/testdata/trustroot/e2e/with-tsa.yaml:21]: I am still working on this ... thanks for looking at it. I am adding a WIP.

Comment (hectorj2f) [reply test/testdata/trustroot/e2e/with-tsa.yaml:21]: I'll complete the tests.

Comment (vaikas) [reply test/testdata/trustroot/e2e/with-tsa.yaml:21]: Oh no worries, was just thinking through my assumptions on how things work :) My thinking has been that there always has to be a CA, but maybe that's wrong.

Comment (hectorj2f) [reply test/testdata/trustroot/e2e/with-tsa.yaml:21]: > Don't we still need the signing CA added? It's currently listed as 'required', so if this is a bad assumption we need to relax it in the trustroot API.

You can sign using the TSA as verification and a raw key without anything else.

Comment (hectorj2f) [reply test/testdata/trustroot/e2e/with-tsa.yaml:21]: We should relax the validation.

Comment (hectorj2f) [reply test/testdata/trustroot/e2e/with-tsa.yaml:22]: Done

File: .github/workflows/kind-cluster-image-policy-tsa.yaml [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+    - name: Install yq
+      uses: mikefarah/yq@87cba2ecbeaecf860efcceb66deab46ae030ce1e # v4.27.3
+
+    - name: Setup mirror
+      uses: chainguard-dev/actions/setup-mirror@main
+      with:
+        mirror: mirror.gcr.io
+
+    - uses: hectorj2f/cosign-installer@c5ac9ce01eb4b0048c02123cb3721624c8f4dc55 # v2
```
Comment (hectorj2f) [inline .github/workflows/kind-cluster-image-policy-tsa.yaml:68]: I am waiting for a decision so we can install cosign v2 and RC versions via https://github.com/sigstore/cosign-installer/pull/105

**Review by mattmoor — APPROVED** [review]
Can you open tracking issues for updating the installer/tsa/cosign deps once they are released with the changes we need?

**Comment by hectorj2f** [comment]
@mattmoor Yes, I'll create one issue. I am working myself on the change too https://github.com/sigstore/cosign-installer/pull/105.

File: .github/workflows/kind-cluster-image-policy-tsa.yaml (hunk 2 above)
Comment (hectorj2f) [reply .github/workflows/kind-cluster-image-policy-tsa.yaml:68]: I created https://github.com/sigstore/policy-controller/issues/478 to track this change.

**Review by priyawadhwa — APPROVED** [review]

**Pull request merged**